bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -56,8 +56,13 @@ public PageChangeListener(String sitemapName, String pageId, ItemUIRegistry item
         this.sitemapName = sitemapName;
         this.pageId = pageId;
         this.itemUIRegistry = itemUIRegistry;
+
+        updateItemsAndWidgets(widgets);
+    }
+
+    private void updateItemsAndWidgets(EList<Widget> widgets) {
         this.widgets = widgets;
-        items = getAllItems(widgets);
+        this.items = getAllItems(widgets);
```

Comment:
What happens if the updated sitemap contains _less_ widgets than before? Please remove this class as a StateChangeListener from the existing items first, otherwise this will leak references!